instrumentation/lettuce/lettuce-5.2/src/main/java8/io/opentelemetry/auto/instrumentation/lettuce/v5_2/OpenTelemetryTracing.java

```diff
@@ -0,0 +1,302 @@
+  }
+
+  @Override
+  public boolean isEnabled() {
+    return true;
+  }
+
+  @Override
+  public boolean includeCommandArgsInSpanTags() {
+    return true;
+  }
+
+  @Override
+  public Endpoint createEndpoint(SocketAddress socketAddress) {
+    if (socketAddress instanceof InetSocketAddress) {
+      InetSocketAddress address = (InetSocketAddress) socketAddress;
+
+      return new OpenTelemetryEndpoint(
+          address.getAddress().getHostAddress(), address.getPort(), address.getHostString());
+    }
+    return null;
+  }
+
+  enum OpenTelemetryTracerProvider implements TracerProvider {
+    INSTANCE;
+
+    private final Tracer openTelemetryTracer = new OpenTelemetryTracer();
+
+    @Override
+    public Tracer getTracer() {
+      return openTelemetryTracer;
+    }
+  }
+
+  enum OpenTelemetryTraceContextProvider implements TraceContextProvider {
+    INSTANCE;
+
+    @Override
+    public TraceContext getTraceContext() {
+      return new OpenTelemetryTraceContext();
+    }
+  }
+
+  static class OpenTelemetryTraceContext implements TraceContext {
+    private final Context context;
+
+    protected OpenTelemetryTraceContext() {
+      this.context = Context.current();
+    }
+
+    public Context getContext() {
+      return context;
+    }
+  }
+
+  static class OpenTelemetryEndpoint implements Endpoint {
+    final String ip;
+    final int port;
+    @Nullable final String name;
+
+    protected OpenTelemetryEndpoint(String ip, int port, @Nullable String name) {
+      this.ip = ip;
+      this.port = port;
+      if (!ip.equals(name)) {
+        this.name = name;
+      } else {
+        this.name = null;
+      }
+    }
+  }
+
+  static class OpenTelemetryTracer extends Tracer {
+
+    protected OpenTelemetryTracer() {}
+
+    @Override
+    public OpenTelemetrySpan nextSpan() {
+      return new OpenTelemetrySpan(null);
+    }
+
+    @Override
+    public OpenTelemetrySpan nextSpan(TraceContext traceContext) {
+      if (!(traceContext instanceof OpenTelemetryTraceContext)) {
+        return nextSpan();
+      }
+
+      Context context = ((OpenTelemetryTraceContext) traceContext).getContext();
+
+      final io.opentelemetry.trace.Span parent = TracingContextUtils.getSpanWithoutDefault(context);
+
+      return new OpenTelemetrySpan(parent);
+    }
+  }
+
+  // The order that callbacks will be called in or which thread they are called from is not well
+  // defined. We go ahead and buffer all data until we know we have a span. This implementation is
+  // particularly safe, synchronizing all accesses. Relying on implementation details would allow
+  // reducing synchronization but the impact should be minimal.
+  static class OpenTelemetrySpan extends Tracer.Span {
+    @Nullable private final Span parent;
+
+    @Nullable private OpenTelemetryEndpoint endpoint;
+    @Nullable private String name;
+
+    @Nullable private List<Object> events;
+    @Nullable private List<String> tags;
```

Comment:
what do u think about replacing `parent`, `endpoint`, `name` and `tags` with `Span.Builder`?